db/index.js

```diff
@@ -837,6 +837,11 @@ function queryTable(store, table, data, opts, isLocal, fetchFromItemDb, startKey
 
   var queryFilter = data.QueryFilter || data.ScanFilter
 
+  var preFilteredItems = []
+  vals.join(function(items) {
+    preFilteredItems = items
```

Comment:
This is happening in a callback, so it's non-deterministic that `preFilteredItems` will be populated by the time the callback down below is called.

I think it's best just to do it all in one callback